app/src/main/java/com/example/mycleaninglog/ExpandableCard.kt

```diff
@@ -204,37 +210,47 @@ fun ExpandableCard(
-                //displays the options for if the card is expanded
-                if(expandedState){
-                    //what to do if the 1st level expandable card is labeled "Rooms"
-                    if (title == "Rooms"){
+                //
+                IsCardExpanded( expandedState, addRoomList, title )
+            }
 
-                        //passes each item on the addRoomList into expandable card level
+        }
+    }
 
-                        //addRoomList.forEach{position -> ExpandableCardLevelTwo(title = position as String)}
-                        //addRoomList.forEach{position -> ExpandableCardLevelTwo(title = position as String)}
-                        addRoomList.forEach{position -> ExpandableCardLevelTwo(title = position.name)}
-                    }
 
-                    //what to do if the 1st level expandable card is labeled "Common Tasks"
-                    if (title == "Common Tasks"){
-                        //currently just displays a text line
-                        //following code needs to be replaced with a list of tasks that are shared between multiple rooms
-                        Text(text = "Here are the common tasks")
-                    }
-                    //what to do if the 1st level expandable card is labeled "Upcoming tasks"
-                    if (title == "Upcoming Tasks"){
-                        //currently just displays a text line
-                        //following code needs to be replaced with a list of tasks in order from most needed to least
-                        Text(text = "here are the upcoming tasks")
-                    }
+@ExperimentalMaterialApi
+@Composable
+// Determines if the card is expanded
+fun IsCardExpanded(expandedState: Boolean, addRoomList: SnapshotStateList<RoomClass>, title: String) {
+    // True?
+    if ( expandedState ){
+        // Do this if the title is 'Rooms'
+        if (title == "Rooms"){
 
-                }
-            }
+            // passes each item on the addRoomList into expandable card level
 
+            //addRoomList.forEach{position -> ExpandableCardLevelTwo(title = position as String)}
+            //addRoomList.forEach{position -> ExpandableCardLevelTwo(title = position as String)}
+            addRoomList.forEach{position -> ExpandableCardLevelTwo(title = position.name)}
+        }
+
+        // Do this if the title is 'Common Tasks'
+        if ( title == "Common Tasks" ){
+            /* following code needs to be replaced with a list of tasks that are
+            *  shared between multiple rooms
+            */
+            Text(text = "Here are the common tasks")
+        }
+        // Do this if the title is 'UpcomingTasks'
+        if ( title == "Upcoming Tasks" ){
+            /* following code needs to be replaced with a list of tasks in order
+             * from most needed to least
+             */
+            Text(text = "here are the upcoming tasks")
         }
-    }
 
+    }
+}
```

Comment:
Added this function and called it in the ExpandableCard function in an effort to make the Expandable Card function smaller. Ultimately I think it makes reading the both functions a lot easier.

Reply:
Smaller, more modular functions are easier to manage.